src/bp-moderation/classes/class-bp-moderation-messages.php

```diff
@@ -52,6 +52,10 @@ public function __construct() {
 
 		// button class.
 		add_filter( 'bp_moderation_get_report_button_args', array( $this, 'update_button_args' ), 10, 3 );
+
+		// Delete message moderation data when message thread is deleted.
+		//add_action( 'bp_messages_thread_after_delete', array( $this, 'delete_moderation_data' ), 10, 4 );
```

Comment:
@janakkaneriya007 Why this action is commented. if it's not needed then remove it. Also, remove `delete_moderation_data` function if not needed